chainer/function.py

```diff
@@ -97,13 +99,20 @@ def __call__(self, *inputs):
             :class:`Variable` objects.
 
         """
+
         in_data = tuple([x.data for x in inputs])
         if self.type_check_enable:
             self._check_data_type_forward(in_data)
+
+        hooks = chainer.global_hooks.values() + self.local_hooks.values()
```

Comment:
local hooks are preceded by global hooks in your comment. This is reversed order of my implementation. Is it intentional?